services/models/istio_rule.go

```diff
@@ -11,14 +11,14 @@ type IstioRuleList struct {
 
 type IstioRule struct {
 	Name    string      `json:"name"`
-	Match   interface{} `json:"match"`
-	Actions interface{} `json:"actions"`
+	Match   interface{} `json:"match,omitempty"`
+	Actions interface{} `json:"actions,omitempty"`
```

Comment:
can you please elaborate how do you decide what can be marked with omitempty and what should not? for instance, Actions here is marked as omitempty, but below in IstioRuleDetails it is not.
I guess it is more clear to whoever is more familiar with those objects than me.
I'm also wondering if we could/should have some schema, perhaps something based on Istio objects to have this perhaps semi automatically detected what can be empty and what is a mandatory field?